third_party/maya/lib/pxrUsdMayaGL/usdTaskDelegate.cpp

```diff
@@ -0,0 +1,456 @@
+//
+// Copyright 2016 Pixar
+//
+// Licensed under the Apache License, Version 2.0 (the "Apache License")
+// with the following modification; you may not use this file except in
+// compliance with the Apache License and the following modification to it:
+// Section 6. Trademarks. is deleted and replaced with:
+//
+// 6. Trademarks. This License does not grant permission to use the trade
+//    names, trademarks, service marks, or product names of the Licensor
+//    and its affiliates, except as required to comply with Section 4(c) of
+//    the License and to reproduce the content of the NOTICE file.
+//
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
+// Some header #define's Bool as int, which breaks stuff in sdf/types.h.
+// Include it first to sidestep the problem. :-/
+#include "pxr/imaging/hdSt/camera.h"
+#include "pxr/imaging/hdSt/light.h"
+
+#include "pxr/imaging/hdx/renderTask.h"
+#include "pxr/imaging/hdx/shadowTask.h"
+#include "pxr/imaging/hdx/shadowMatrixComputation.h"
+#include "pxr/imaging/hdx/selectionTask.h"
+#include "pxr/imaging/hdx/simpleLightTask.h"
+
+#include "pxr/base/gf/frustum.h"
+#include "pxr/base/tf/staticTokens.h"
+
+#include "usdTaskDelegate.h"
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+TF_DEFINE_PRIVATE_TOKENS(
+	_tokens,
+	(idRenderTask)
+	(renderTask)
+	(shadowTask)
+	(selectionTask)
+	(simpleLightTask)
+	(camera)
+);
+
+namespace {
+	class ShadowMatrix : public HdxShadowMatrixComputation
```

Comment:
This is awesome that you're tackling this too! I was actually playing with something similar internally, but I hadn't yet finished it. I **think** that the camera needs a windowPolicy set on it though for this to work, and I don't see that happening down below. Was that left out of this change, maybe?

Also, a few of the values used to setup the frustum look like they're hard-coded and should likely instead be derived from properties of the light.